evm_loader/lib/src/types/tracer_ch_db.rs

```diff
@@ -244,24 +251,35 @@ impl ClickHouseDb {
         let mut row = if branch.is_empty() {
             None
         } else {
-            let query = r#"
-                SELECT owner, lamports, executable, rent_epoch, data, txn_signature
-                FROM events.update_account_distributed
-                WHERE pubkey = ?
-                  AND slot IN ?
-                ORDER BY pubkey, slot DESC, write_version DESC
-                LIMIT 1
-            "#;
+            let query = if write_version.is_some() {
+                r#"
+                    SELECT owner, lamports, executable, rent_epoch, data, txn_signature
+                    FROM events.update_account_distributed
+                    WHERE pubkey = ?
+                      AND write_version = ?
```

Comment:
`write_version` is valid only for the `slot` specified as an argument to `get_account_at()` call. So the algorithm should first look for an `update_account` event into the current `slot = ?` and `write_version <= ?` (bound to `slot` and `write_version`), then fall back to the previous implementation of finding an update_account event into parent slots. I added a suggestion for this here: https://github.com/neonlabsorg/neon-evm/pull/213

Reply:
Makes sense, I have updated the pull request. 👍